src/waveform/renderers/waveformrendermarkrange.cpp

```diff
@@ -45,18 +48,20 @@ void WaveformRenderMarkRange::draw(QPainter *painter, QPaintEvent * /*event*/) {
         generateImages();
     }
 
-    for (unsigned int i = 0; i < m_markRanges.size(); i++) {
-        WaveformMarkRange& markRange = m_markRanges[i];
+    for (auto const& markRange: m_markRanges) {
+        if (!markRange) {
```

Comment:
No pointer in the member m_markRanges is ever invalidated. This case is impossible and must be a debug assertion.

Reply:
Ah OK I'll change this now.